src/Phpbb/TranslationValidator/Command/DownloadCommand.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ *
+ * @package phpBB Translation Validator
+ * @copyright (c) 2021 phpBB Ltd.
+ * @license http://opensource.org/licenses/gpl-2.0.php GNU General Public License v2
+ *
+ */
+
+namespace Phpbb\TranslationValidator\Command;
+
+use Phpbb\TranslationValidator\Output\Output;
+use Phpbb\TranslationValidator\Output\OutputFormatter;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DownloadCommand extends Command
+{
+	const GITHUB_API_URL = 'https://api.github.com/repos/%s/git/trees/master?recursive=1';
+	const GITHUB_LANGUAGE_EXTRACT = 'language/en/';
+
+	// Supported extensions
+	const VIGLINK_EXTENSION = 'phpbb-extensions/viglink';
+	const VIGLINK_PATH = 'ext/phpbb/viglink/';
+
+	protected function configure()
+	{
+		$this
+			->setName('download')
+			->setDescription('If you are missing important files, this tool can automatically download them for you.')
+			->addOption('files', null, InputOption::VALUE_REQUIRED, 'Which files do you want to download?', 'phpbb-extensions/viglink')
+			->addOption('phpbb-version', null, InputOption::VALUE_OPTIONAL, 'The phpBB version you use to validate against', '3.3');
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		$files = $input->getOption('files');
+		$phpbbVersion = $input->getOption('phpbb-version');
+
+		if (!in_array($files, [self::VIGLINK_EXTENSION]))
+		{
+			throw new \RuntimeException($files . ' is not supported for automatic download.');
+		}
+
+		$output = new Output($output, false);
+		$output->setFormatter(new OutputFormatter($output->isDecorated()));
+
+		$output->writeln('Downloading ' . $files);
+
+		if ($files === self::VIGLINK_EXTENSION)
+		{
+			// Download Viglink files if they are missing
+			$this->downloadViglinkExtensionLanguagesFiles($output, $phpbbVersion);
+		}
+
+		$output->writeln('Script complete.');
+	}
+
+	/**
+	 * Download missing Viglink files and store them in phpbb-translation-validator/3.x/en/ext/phpbb/viglink/language/en
```

Comment:
Is that a relative path? So does this work online and offline on a local maschine?

Reply:
Yep, it is a relative path - I just put phpbb-translation-validator in the comments because that's the name of this repo. If someone changed it locally it would still work though:

```php
$directory = __DIR__ . '/../../../../' . $phpbbVersion . '/en/' . self::VIGLINK_PATH;
```